### PR TITLE
Updates to FE Courses Model + Serializer; Includes URLs & Views Updates

### PR DESCRIPTION
**Pull request by rbassot**
Completes ticket #75 .

Required changes to the `Course` model for the FE team to follow their spec. Changes include:

- Model fields updated & renamed for clarity
- Model primary key changed to unique `course_code`, due to the changes around sections
- Serializers updated to handle model changes
- URLs updated to handle the new PK --> changed UPDATE/DELETE actions to hit: `POST api/courses/{course_code}/`
- Views logic updated to handle the new PK
- Tests updated

Kindly assure that the (new URLs) endpoints can still be hit, and can accept the new Course data format in a request body. The test suite should pass without errors

### CODE REVIEW

**Comment by skosman** [comment]
When you created the new initial migration script, how did you do it? Because when I try to run `migrate`, it does not recognize the new initial migration script so the changes are not applied to the db.

**Comment by rbassot** [comment]
> When you created the new initial migration script, how did you do it? Because when I try to run `migrate`, it does not recognize the new initial migration script so the changes are not applied to the db. It just says no migrations to apply.

Hey - I think if you torch the `/data` directory and rebuild the app + DB entirely with `compose up --build`, then perform `manage.py migrate`, it should work in theory. This is what I did and changes were applied

**Comment by skosman** [comment]
> > When you created the new initial migration script, how did you do it? Because when I try to run `migrate`, it does not recognize the new initial migration script so the changes are not applied to the db. It just says no migrations to apply.
> 
> Hey - I think if you torch the `/data` directory, delete all Docker containers, and rebuild the app + DB entirely with `compose up --build`, then perform `manage.py migrate`, it should work in theory. This is what I did and changes were applied

Okay, that's what I thought. My only concern with that is that I think we'll have to drop the prod database as well to apply the new migration. I'm not sure if that's easy or not, @kunnapool would know better. To keep things simple, I would recommend keeping any database changes as separate migration file to prevent having to torch our prod db.

**Comment by rbassot** [comment]
> > > When you created the new initial migration script, how did you do it? Because when I try to run `migrate`, it does not recognize the new initial migration script so the changes are not applied to the db. It just says no migrations to apply.
> > 
> > 
> > Hey - I think if you torch the `/data` directory, delete all Docker containers, and rebuild the app + DB entirely with `compose up --build`, then perform `manage.py migrate`, it should work in theory. This is what I did and changes were applied
> 
> Oh okay thank you! My only concern with that is that I think we'll have to drop the prod database as well to apply the new migration. I'm not sure if that's easy or not, @kunnapool would know better. To keep things simple, I would recommend keeping any database changes as separate migration file to prevent having to torch our prod db.

I'm pretty sure you're correct, and I will check with Kunal on how this can be implemented

**Comment by kunnapool** [comment]
> > > When you created the new initial migration script, how did you do it? Because when I try to run `migrate`, it does not recognize the new initial migration script so the changes are not applied to the db. It just says no migrations to apply.
> > 
> > 
> > Hey - I think if you torch the `/data` directory, delete all Docker containers, and rebuild the app + DB entirely with `compose up --build`, then perform `manage.py migrate`, it should work in theory. This is what I did and changes were applied
> 
> Oh okay thank you! My only concern with that is that I think we'll have to drop the prod database as well to apply the new migration. I'm not sure if that's easy or not, @kunnapool would know better. To keep things simple, I would recommend keeping any database changes as separate migration file to prevent having to torch our prod db.

Hhmm, we can try to run the migrations, but since it's likely to fail, I can just drop this instance and create a new one. Shouldn't be too bad, I'll just have to be a bit careful with the Security groups and permissions to make sure they're exactly the same as this one. That plus updating the hostname + testing should be ~1 hour

File: courses/tests.py [hunk 1]
```diff
@@ -9,22 +9,22 @@
...
         self.course_attributes = {
             "course_code": "SENG499",
-            "section": "A01",
+            "num_sections": 2,
             "course_title": "Design Project 2",
             "fall_offering": True,
             "spring_offering": True,
             "summer_offering": False,
-            "PENG_required": True,
+            "pengRequired": True,
```
Comment (skosman) [inline courses/tests.py:23]: Smallest thing. Might want to update the pengRequired fields in the tests to match the expected JSON field. Like the 
comment you left on the courses serializer, ex {"fall": true, "spring": false, "summer": true}

**Review by skosman — APPROVED** [review]
Got all the courses endpoints working! Nice job!

File: courses/tests.py (hunk 1 above)
Comment (rbassot) [reply courses/tests.py]: Good catch, thanks for finding this! Fixed.

**Comment by rbassot** [comment]
Looks like all Courses-affected tests are now passing 💯

**Pull request merged**